staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -227,18 +252,58 @@ func (s *Controller) Run(stopCh <-chan struct{}, workers int) {
 		go wait.Until(s.worker, time.Second, stopCh)
 	}
 
-	go wait.Until(s.nodeSyncLoop, nodeSyncPeriod, stopCh)
+	go s.nodeSyncLoop(workers)
+	go wait.Until(s.triggerNodeSync, nodeSyncPeriod, stopCh)
 
 	<-stopCh
 }
 
+// triggerNodeSync triggers a nodeSync asynchronously
+func (s *Controller) triggerNodeSync() {
+	s.triggerLock.Lock()
+	defer s.triggerLock.Unlock()
+	newHosts, err := listWithPredicate(s.nodeLister, s.getNodeConditionPredicate())
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("Failed to retrieve current set of nodes from node lister: %v", err))
+		// if node list cannot be retrieve, trigger full node sync to be safe.
+		s.markFullSync()
+	} else if !nodeSlicesEqualForLB(newHosts, s.knownHosts) {
+		// Here the last known state is recorded as knownHosts. For each
+		// LB update, the latest node list is retrieved. This is to prevent a stale set of nodes were used to be update
```

Comment:
nit: can you please split those lines to make them shorter?

Reply:
Done